### PR TITLE
Update the installation guide

### LINKED ISSUE #96: Add details about why users want each package to framework installation instructions
### Issue Description
We have redundant instructions on [this site](https://docs.signac.io/en/latest/installation.html) but also [signac-core](https://docs.signac.io/projects/core/en/latest/installation.html) and [signac-flow](https://docs.signac.io/projects/flow/en/latest/installation.html) and [signac-dashboard](https://docs.signac.io/projects/dashboard/en/latest/installation.html) (all links are to "installation".)

The installation tab in signac-docs clutters our sidebar and detracts from the purpose of the integrated documentation, which is to show how the pieces fit together. 

### Proposed change

We had discussed removing the installation instructions in this week's signac dev meeting. Keep installation instructions only within each package. Link to each installation page directly from the getting started page and the package list page.

### PR DESCRIPTION
**Pull request by kidrahahjo**
## Description
This pull request updates the docs to provide an explanation to the users of why they should install the packages in the signac framework.

## Motivation and Context
Resolves #96 

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-docs/blob/master/ContributorAgreement.md).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-docs/blob/master/CONTRIBUTING.md#code-style) of this project.

### CODE REVIEW

File: docs/source/installation.rst
Comment (bdice) [inline docs/source/installation.rst:30]:
```suggestion
- Share workspaces through a browser-based GUI.
- Visualize job data such as text, images, or video.
- Requires the core **signac** package.
```

File: docs/source/installation.rst
Comment (bdice) [inline docs/source/installation.rst:18]:
```suggestion
- Manage project data with a well-defined indexable storage layout for data and metadata.
- Search and manipulate the data in existing **signac** projects.
- Provides the data management functionality for **signac-flow** and **signac-dashboard**.
```

File: docs/source/installation.rst
Comment (bdice) [inline docs/source/installation.rst:25]:
```suggestion
- Implement workflows for projects managed with **signac**.
- Run complex workflows from the command line.
- Submit jobs to high-performance clusters (HPC).
- Requires the core **signac** package.
```

File: docs/source/installation.rst
Comment (bdice) [inline docs/source/installation.rst:12]:
```suggestion
All packages in the **signac** framework depend on the core **signac** package, which provides the data management functionality used by all other packages.
```

**Review by bdice — CHANGES_REQUESTED** [review]
This duplicates a fair bit of the content in [packages.rst](https://github.com/glotzerlab/signac-docs/blob/master/docs/source/packages.rst). I wonder if we should combine this information.

**Comment by kidrahahjo** [comment]
@bdice I believe it's a bit easier (for the eyes as well) to look at the information in a structured sense (Not sure though).
I think duplication up to a little extent is okay (maybe?)
Please suggest.

**Comment by vyasr** [comment]
I think I agree with @kidrahahjo that the bulleted format (what I assume he means by a "structured sense") is helpful to summarize. However, if we agree on that, I would advocate for changing packages.rst instead and linking to it. While I would avoid rewriting the same thing in two places, I don't mind duplication in the rendered docs, so we could modify packages.rst (if we think that does look better) than just `literalinclude` it here.

**Comment by kidrahahjo** [comment]
I think I misinterpreted @bdice's review of combining the information.
Since both @bdice and @vyasr are in favour of not duplicating this info. 
I'll combine the information.

**Comment by cbkerr** [comment]
Thank you for working on this @kidrahahjo! I also agree with @vyas, I like the bulleted form of the description of each package. However, I think having a `literalinclude` in the installation page seems a bit excessive.

Could you add the bulleted lists removed from "Installation" in commit 9ac4cba6e1b5181e40ee728c2f31938a792d7f92 to `packages.rst`? These bulleted lists would replace the paragraphs of text that are currently in "Packages" describing each package.

**Review by bdice — APPROVED** [review]
@vyasr @cbkerr @kidrahahjo I pushed some changes to this branch based on the discussion above. I think this is ready to merge. If anyone would like to weigh in or edit this branch directly, feel free to do so. We can merge this once it receives a second approval.

File: docs/source/packages.rst
Comment (cbkerr) [inline docs/source/packages.rst:54]:
```suggestion
The **signac-dashboard** package allows users to browse **signac**-managed data spaces:
```
removed bc redundant in bullet

File: docs/source/packages.rst
Comment (cbkerr) [inline docs/source/packages.rst:40]:
```suggestion
The **signac-flow** package allows users to:

- Implement workflows for a project data space managed with **signac**.
```

trying to remove redundancy

File: docs/source/packages.rst
Comment (cbkerr) [inline docs/source/packages.rst:23]:
```suggestion
The **signac** package allows users to:
```

reframing with focus on what the users wants

File: docs/source/packages.rst
Comment (cbkerr) [inline docs/source/packages.rst:27]:
```suggestion
- Unlock the use of **signac-flow** and **signac-dashboard** for additional utility.
```
Phrase the dependency as what the user gets out of it

**Review by cbkerr — CHANGES_REQUESTED** [review]
Overall this looks great! I made a few suggestions to focus on what the user gets out of each package

File: docs/source/packages.rst
Comment (kidrahahjo) [reply docs/source/packages.rst:27]: I'm not sure if unlock is the right word here, it might confuse the users that the core signac package is dependent on signac-flow and signac-dashboard. What do you think about this?

File: docs/source/packages.rst
Comment (cbkerr) [reply docs/source/packages.rst]: added a new commit to fix first bullet redundancy

File: docs/source/packages.rst
Comment (cbkerr) [reply docs/source/packages.rst:27]: I want to show that signac-flow and dashboard depend on signac. But I also want to keep a parallel grammatical structure to what the user can do. Maybe enable instead of unlock?
"Enable the use of signac-flow and signac-dashboard " ?
As in: you have to use signac to enable the other packages.

**Actually, I think we can just get rid of this last bullet -- dependencies can be dealt with on the installation page** Even with these revisions it doesn't tell me much about what the user gets out of the package

Comment (kidrahahjo) [reply docs/source/packages.rst:27]: To show the dependency between these packages, we mention that _"You only have to install those that provide the functionality you need, however both **signac-flow** and **signac-dashboard** require the **signac** core package."_ in the introductory paragraph of `packages` page

Comment (cbkerr) [reply docs/source/packages.rst:27]: Good point! I felt safe to remove that bullet then

File: docs/source/packages.rst
Comment (bdice) [inline docs/source/packages.rst:39]:
```suggestion
- Implement reproducible computational workflows for a project data space managed with **signac**.
```

**Review by bdice — CHANGES_REQUESTED** [review]

**Comment by bdice** [comment]
@cbkerr @kidrahahjo This PR can be merged once there is consensus on what edits should be made. I'd like to avoid having many more rounds of feedback here, in the interest of moving this PR forward.

**Review by cbkerr — APPROVED** [review]

**Comment by cbkerr** [comment]
I agree with the rest of @bdice's suggestions and revised a few wordings. I think this PR improves the readability of the documentation and is ready to merge.

**Review by bdice — APPROVED** [review]
Great, I'll merge it. Thanks @cbkerr @kidrahahjo @vyasr for the input. 👍

**Pull request merged**
